support-frontend/assets/helpers/paymentIntegrations/readerRevenueApis.js

```diff
@@ -166,7 +169,8 @@ export type ExistingDirectDebitAuthorisation = {|
 |};
 export type AmazonPayAuthorisation = {|
   paymentMethod: typeof AmazonPay,
-  orderReferenceId: string
+  orderReferenceId?: string,
+  amazonPayBillingAgreementId?: string
```

Comment:
It's not ideal to be using this type for both one-off and recurring. This payment authorisation logic could do with a refactor to split one-off and recurring